### ISSUE TITLE
Bug: Value {"ignoredArrowShorthand":true} should NOT have additional properties

### ISSUE THREAD

**Issue by mkosir** [typescript-eslint/typescript-eslint#7217]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Relevant Package

parser

### Repro Code

```typescript
'@typescript-eslint/no-confusing-void-expression': ['error', { ignoredArrowShorthand: true }],
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
 extends: [
    'plugin:@typescript-eslint/recommended-type-checked',
    'plugin:@typescript-eslint/strict-type-checked',
    'plugin:@typescript-eslint/stylistic-type-checked',
  ],
  rules: {
    '@typescript-eslint/no-confusing-void-expression': ['error', { ignoredArrowShorthand: true }],
  },
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    // ...
  }
}
```

### Expected Result

When running lint ESlint v8.44.0 (latest) with typescript-eslint v6.0.0 (latest) reports an error:
```sh
ESLint: 8.44.0

Error: .eslintrc.js:
        Configuration for rule "@typescript-eslint/no-confusing-void-expression" is invalid:
        Value {"ignoredArrowShorthand":true} should NOT have additional properties.
```

### Actual Result

Linting should be executed without any errors.

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin`     | `6.0.0` |
| `@typescript-eslint/parser`            | `6.0.0` |
| `TypeScript`                           | `5.1.6` |
| `ESLint`                               | `8.44.0` |
| `node`                                 | `18.16.1` |

**Comment by Josh-Cena** [comment]
The option is called `ignoreArrowShorthand`; you have an extra `d`, which is why it's not recognized, hence, an "extra property".

**Comment by JoshuaKGoldberg** [comment]
...I wonder if we could file an issue on ESLint do add _"did you mean <...>?"_ suggestions to these...?

**Comment by mkosir** [comment]
thanks! will stop coding for today 🙈

**Comment by mkosir** [comment]
Well since I already opened an issue, small shoutout, just trying out and upgrading projects to new `v6` - it's awesome! Keep up the great work 🚀

**Comment by JoshuaKGoldberg** [comment]
Yay, thanks mkosir!! 😍 

It's honestly a relief to hear that.
